src/extra/libs/png/lv_png.c

```diff
@@ -24,8 +24,9 @@
 /**********************
  *  STATIC PROTOTYPES
  **********************/
-static lv_res_t decoder_info(struct _lv_img_decoder_t * decoder, const void * src, lv_img_header_t * header);
-static lv_res_t decoder_open(lv_img_decoder_t * dec, lv_img_decoder_dsc_t * dsc);
+static lv_res_t decoder_info(struct _lv_img_decoder_t * decoder, const void * src, lv_img_header_t * header,
+                             void * dec_ctx);
+static lv_res_t decoder_open(lv_img_decoder_t * dec, lv_img_decoder_dsc_t * dsc, void * dec_ctx);
```

Comment:
Why not pass the ctx to `decoder_close` to close too?

Reply:
The context is moved to the descriptor in open. So it's available there, no need to pass it again (neither in read_line). Also, the decoder can be closed while dealing with another picture (and so another decoder context) when the cache is pruned so we don't have the context anymore.

Reply:
Ah, I see now. In `open` you save `dec_ctx` in `context` and `context` in the decoder's `user_data`. 

IMO as `dec_ctx` has its own meaning we should add a new filed for in the `lv_img_decoder_dsc_t` for it and leave `user_data` for more general and arbitrary stuff. E.g. the MicroPython binding uses the `user_data` to attach some data there.

Reply:
Yes exactly. Let's do that.

Reply:
Done.